packages/clerk-js/src/core/clerk.ts

```diff
@@ -250,19 +256,36 @@ export default class Clerk implements ClerkInterface {
     });
   };
 
-  public setSession = async (
-    session: ActiveSessionResource | string | null,
-    beforeEmit?: BeforeEmitCallback,
-  ): Promise<void> => {
+  public setActive = async ({ session, organization, beforeEmit }: SetActiveParams): Promise<void> => {
```

Comment:
I believe this method could probably be written much simpler if we re-arrange the code to have some early returns.
However, given how critical this is, I hesitated to do it.

Reply:
@alex-ntousias 
I agree and the thing is there so many sideEffect functions which makes it daunting.

Reply:
cc: @SokratisVidros @nikosdouvlis @raptisj ☝️

Reply:
I agree. We will do that separately. There is a relevant [epic](https://www.notion.so/clerkdev/Revamp-ClerkJS-setSessionz-9def3ea6a35b464baa69ab9f83bc5c91).